n3fit/src/n3fit/model_trainer.py

```diff
@@ -355,6 +359,16 @@ def _model_generation(self, pdf_models, partition, partition_idx):
             # Apply feature scaling if given
             input_arr = self._scaler(input_arr)
         input_layer = op.numpy_to_input(input_arr)
+        
+        photon_list = [
+            np.array([photon_fitting_scale(
+                xgrid=grid[0],
+                theoryid=N3FIT_FIXED_CONFIG['theoryid'],
+                fiatlux_runcard = N3FIT_FIXED_CONFIG['fiatlux'],
```

Comment:
This information should enter as arguments to this function. Don't import the config from `n3fit`.

Also, since this is a function that is supposed to create the photon at the fitting scale, I'd rather have something like this:

```python

class Photon:
    def __init__(theoryid, fiatlux_runcard):
         ... do some initialization ...
    
     def photon_fitting_scale(xgrid):
           ... generate the photon and only do that ...
```

this is just an example. Not sure where to put the replica, I guess down there: `photon_fitting_scale(xgrid)`.

The reason for this is that I image we might want to compute the photon by itself within the nnpdf code at some point and then we will need to extract it from `n3fit`. This has already happened so many times that I'd rather have things outside now :__

Reply:
I'm not sure we need an entire class: this seems to me a bad idea, since the work done here is minimal, everything actually happens inside `fiatlux`.
I'd be in favor of using just a function:
```py
def photon_fitting_scale(xgrid):
    ...
```

Reply:
But you need the fiatlux information right? (but ok, the class is just one example, is just that `n3fit` should not get any information about fiatlux, it needs only the photon)

Reply:
Indeed, I agree with you: but we just need to get the array and that's it. Let's simply try to keep it minimal